### ISSUE TITLE
Unable to detect second object

### ISSUE THREAD

**Issue by Ram124** [ijkguo/mx-rcnn#105]
Hi @ijkguo ,
I have trained the model with resnet18.
I have given 2 classes. 
when i run test.py. After 10 epochs
I am getting something like this..

INFO:root:Writing pedestrian VOC results file
INFO:root:Writing bicycle VOC results file
INFO:root:AP for __background__ = 0.5177
INFO:root:AP for pedestrian = 0.2642
INFO:root:Mean AP = 0.3910

Unable to detect 2nd class. 
What is the problem??

There is something wrong in the train.py?
Please clarify this.. 
Thanks in advance.

**Comment by ijkguo** [comment]
Your custom classes seem to be background (index 0), pedestrian (index 1), bicycle (index 2). Please check the output labels to be 0, 1 or 2.

**Comment by Ram124** [comment]
@ijkguo 
I have changed the classes names in pascal_voc.py .. 
I have included '_background_', 'pedestrian', 'bicycle'. 
Changed the class numbers to 3 in train and test, demo scripts. 
I have used v6 branch...

It should work right? It is not happening..
Should i change anything else??

**Comment by ijkguo** [comment]
https://github.com/ijkguo/mx-rcnn/blob/dff47b8c4bee1121ce74414ec06f170eafac06e1/symimdb/pascal_voc.py#L165
the background class was not skipped when printing the evaluation results, thus background was referring to the next class. In your case, 0.5177 is AP for pedestrian and 0.2642 is AP for bicycle. I made a fix here: https://github.com/ijkguo/mx-rcnn/commit/612d7c97f33031df3ceffb511042e30e57f4a2d1.
Thanks for reporting it: nothing was wrong with your training.

**Comment by Ram124** [comment]
I had thought the same.. 
i changed it and checked. 
But when i run on some demo images 200+, 
It is only detecting pedestrian, bicycle is not getting detected.. 
even if i run 25 epochs, it is not detecting..
I am completely stuck here from past few days..

**Comment by ijkguo** [comment]
OK so it is a different problem now. Based on https://github.com/ijkguo/mx-rcnn/issues/95, you have 200 images in your training set, 200 images in your testing set. I think more data is better for generalization.

**Comment by Ram124** [comment]
No, i have 1000 images of each class now..
but still i am not able to detect the second object.

**Comment by ijkguo** [comment]
Let's look at training set first before evaluating generalization ability on testing set. Can you see more than one detection in a training image?

**Comment by Ram124** [comment]
I have training set with sigle object in each image. So only one bounding
box per image.

But while doing inference, i have dataset of images with 2 targets in
single image..

But if i do train the network n then test them,
Both the objects are detected as Pedestrian itself.
Even if single object bicycle present sometimes,
It is also detecting as pedestrian.
